.github/workflows/regression-tests.yml

```diff
@@ -0,0 +1,472 @@
+            const matches = output.matchAll(new RegExp(process.env.PATTERN, 'g'));
+            const sums = [];
+            for (const groups of matches) {
+              for (let i = 1; i < groups.length; i++) {
+                while (sums.length < i) {
+                  sums.push(0);
+                }
+                sums[i - 1] += parseInt(groups[i]);
+              }
+            }
+            let details = process.env.TEMPLATE;
+            for (let i = 0; i < sums.length; i++) {
+              details = details.replaceAll(`\${${i}}`, sums[i]);
+            }
+            fs.writeFileSync('run.details', details);
+      - id: upload
+        env:
+          REPOSITORY: ${{ matrix.repository }}
+          CONTEXT: |
+            {
+              "repository": "${{ matrix.repository }}",
+              "runner": "${{ matrix.runner }}",
+              "command": "${{ matrix.command }}"
+            }
+        run: |
+          echo "repository=${REPOSITORY//\//_}" | tee -a $GITHUB_OUTPUT
+          echo "$CONTEXT" > run.context
+      - name: Upload the result
+        uses: actions/upload-artifact@v4
+        with:
+          name: repository-${{ steps.upload.outputs.repository }};command-${{ matrix.command }};runner-${{ matrix.runner }}
+          path: |
+            run.context
+            run.output
+            run.result
+            run.details
+
+  summarize:
+    name: Summarize the test results
+    if: (!cancelled())
+    needs: [test]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download the results
+        uses: actions/download-artifact@v4
+        with:
+          pattern: 'repository-*;command-*;runner-*'
+      - name: Summarize the results
+        id: summary
+        env:
+          HARDHAT_REF: ${{ inputs.hardhat-ref }}
+          EDR_REF: ${{ inputs.edr-ref }}
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const fs = require('fs');
+            const path = require('path');
+            const artifacts = fs.readdirSync(process.env.GITHUB_WORKSPACE);
+            const results = [];
+            for (const artifact of artifacts) {
+              const context = JSON.parse(fs.readFileSync(path.join(artifact, 'run.context')).toString());
+              const result = JSON.parse(fs.readFileSync(path.join(artifact, 'run.result')).toString());
+              const details = fs.readFileSync(path.join(artifact, 'run.details')).toString();
+              results.push({
+                repository: context.repository,
+                command: context.command,
+                runner: context.runner,
+                result: result.status,
+                time: result.time,
+                details,
+              });
+            }
+            console.log(results);
+            const header = ['Repository 📦', 'Command 👾', 'Runner 💨', 'Result 🧪', 'Wall-clock Time ⏰', 'Details 🔎'].map(data => ({data, header: true}));
```

Comment:
I included the command run-time in the output because I think it will still be useful for us to get a rough idea of how long it took, even accounting for hosted runners instability. However, to address the issues raised in the design doc, I decided not to run forge commands by defaults to remove the notion of us comparing ourselves to them in terms of runtime based on the results of this workflow.